sky/backends/backend_utils.py

```diff
@@ -138,36 +137,37 @@ def fill_template(template_name: str,
         output_path = str(output_path)
     output_path = os.path.abspath(output_path)
 
-    # Runtime files handling
-    #
-    # List of runtime files to be uploaded to cluster:
-    #   - yaml config (for autostopping)
-    #   - wheel
-    #   - credentials
-    # Format is {dst: src}.
-    file_mounts = {SKY_RAY_YAML_REMOTE_PATH: output_path}
-
-    # fill_template() is also called to fill TPU/spot controller templates,
-    # which don't have all variables.
-    if 'sky_remote_path' in variables and 'sky_local_path' in variables:
-        file_mounts[variables['sky_remote_path']] = variables['sky_local_path']
-    if 'credentials' in variables:
-        file_mounts.update(variables['credentials'])
+    # Add yaml file path to the template variables.
+    variables['sky_ray_yaml_remote_path'] = SKY_RAY_YAML_REMOTE_PATH
+    variables['sky_ray_yaml_local_path'] = output_path
+    # Write out yaml config.
+    template = jinja2.Template(template)
+    content = template.render(**variables)
+    with open(output_path, 'w') as fout:
+        fout.write(content)
+    return output_path
+
+
+def _optimize_file_mounts(yaml_path: str):
+    """Optimize file mounts in the given ray yaml file."""
+    with open(yaml_path, 'r') as f:
+        yaml_config = yaml.safe_load(f)
```

Comment:
`yaml_config = common_utils.read_yaml(yaml_path)`? (This suggests that helper func is too shallow - it's as easy to just load the yaml. So maybe we should get rid of that. For now I'd vote for consistency.)

Reply:
Good point! Replaced it with the function. Thanks!